### PR TITLE
SDIT-405: ♻️  Split out index and reindex functions

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/model/translator.kt [hunk 1]
```diff
@@ -8,7 +8,7 @@ import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.canonicalPNCN
 import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.canonicalPNCNumberShort
 import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.dto.nomis.OffenderBooking
 
-fun Prisoner.translate(existingPrisoner: Prisoner? = null, ob: OffenderBooking, incentiveLevel: Result<IncentiveLevel?>, restrictedPatientData: RestrictedPatient?): Prisoner {
+fun Prisoner.translate(existingPrisoner: Prisoner? = null, ob: OffenderBooking, incentiveLevel: Result<IncentiveLevel?>, restrictedPatientData: Result<RestrictedPatient?>): Prisoner {
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/model/translator.kt:11]: this was odd previously - we coped with incentives failing, but not restricted patients

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/services/MaintainIndexService.kt [hunk 2]
```diff
@@ -181,13 +181,12 @@ class MaintainIndexService(
...
-
-  private fun doUpdatePrisoner(indexStatus: IndexStatus, prisonerNumber: String) =
-    with(indexStatus.activeIndexes()) {
-      log.info("Updating prisoner {} on indexes {}", prisonerNumber, this)
-      prisonerSynchroniserService.synchronisePrisoner(prisonerNumber, *this.toTypedArray())
-    }
+      .flatMap {
+        with(it.activeIndexes()) {
+          log.info("Updating prisoner {} on indexes {}", prisonerNumber, this)
+          prisonerSynchroniserService.reindex(prisonerNumber, *this.toTypedArray())
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/services/MaintainIndexService.kt:187]: leaving this as is for now - next step will include logic on whether prisoner still exists in NOMIS etc.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/services/PrisonerSynchroniserServiceTest.kt [hunk 3]
```diff
@@ -46,30 +50,276 @@ internal class PrisonerSynchroniserServiceTest {
...
     @Test
     internal fun `will send telemetry event for update`() {
-      service.synchronisePrisoner("X12345", GREEN)
+      service.reindex("X12345", GREEN)
 
       verify(telemetryClient).trackEvent(eq("PRISONER_UPDATED"), any(), isNull())
     }
+
+    @Test
+    internal fun `will not call restricted patients if prisoner in a prison`() {
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/services/PrisonerSynchroniserServiceTest.kt:80]: turns out that missing lots of tests in the synchroniser service

**Review by lbennett-moj — APPROVED** [review]
lgtm

**Pull request merged**